rust/src/datatypes.rs

```diff
@@ -312,6 +394,40 @@ mod tests {
         }
     }
 
+    #[test]
```

Comment:
can you add a test case for list-of-list-of-float (just to make sure bounding boxes would work well?

Reply:
Done